### DISCUSSION TITLE
Running mitmproxy from code

### DISCUSSION THREAD

**Discussion by ghost** [mitmproxy/mitmproxy#6640] [Intercept Everything!]
I am writing a simple python code using mitmproxy that will intercept all of the https requests that are being sent from my pc and will print their content, I have a simple addon that I wrote but I cant understand how to start the mitmproxy from code like every time I try I get different kinds of errors can someone help?
![image](https://github.com/mitmproxy/mitmproxy/assets/118697863/65da045b-f750-47e9-bcbd-645dbd27446b)

**Comment by changsin** [comment]
You can take a look at DumpMaster (https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/tools/dump.py)
I was able to do it by doing something like this (assuming that your addon class is "your_add_on_class":
    master_instance = dump.DumpMaster(
        opts,
        with_termlog=True,
        with_dumper=False,
    )
    master_instance.addons.add(your_add_on_class)
    await master_instance.run()

**Reply by ghost** [reply]
I tried it but no matter how much docs I read or changes I made to the code it always gave me different errors
`from mitmproxy import addons
from mitmproxy import master
from mitmproxy import options
from mitmproxy.addons import dumper
from mitmproxy.addons import errorcheck
from mitmproxy.addons import keepserving
from mitmproxy.addons import readfile
from mitmproxy import http
import asyncio

class Https:
    def request(flow: http.HTTPFlow) -> None:
        print(flow.request.content)

class DumpMaster(master.Master):
    def __init__(
        self,
        options: options.Options,
        loop=None,
        with_dumper=True,
    ) -> None:
        super().__init__(options, event_loop=loop)
        self.addons.add(*addons.default_addons())
        if with_dumper:
            self.addons.add(dumper.Dumper())
        self.addons.add(
            keepserving.KeepServing(),
            readfile.ReadFileStdin(),
            errorcheck.ErrorCheck(),
        )
async def main():
    opts = options.Options()
    master_instance = DumpMaster(
        opts,
        with_termlog=True,
        with_dumper=False,
    )
    master_instance.addons.add(Https)
    await master_instance.run()


asyncio.run(main())`
